phalcon/db/adapter/pdo.zep

```diff
@@ -360,7 +360,8 @@ abstract class Pdo extends Adapter
 				let statement = this->executePrepared(statement, bindParams, bindTypes);
 			}
 		} else {
-			let statement = pdo->query(sqlStatement);
+			let statement = pdo->prepare(sqlStatement);
+			statement->execute();
 		}
 
```

Comment:
We can reuse existing code here and do some minor refactoring to use `prepare` for both cases.

The code can be refactored as follows:
```php
	public function query(string! sqlStatement, var bindParams = null, var bindTypes = null) -> <ResultInterface> | boolean
	{
		var eventsManager, pdo, statement, params, types;

		let eventsManager = <ManagerInterface> this->_eventsManager;

		/**
		 * Execute the beforeQuery event if an EventsManager is available
		 */
		if typeof eventsManager == "object" {
			let this->_sqlStatement = sqlStatement,
				this->_sqlVariables = bindParams,
				this->_sqlBindTypes = bindTypes;
			if eventsManager->fire("db:beforeQuery", this) === false {
				return false;
			}
		}

		let pdo = <\Pdo> this->_pdo;
		if typeof bindParams == "array" {
			let params = bindParams;
			let types = bindTypes;
		} else {
			let params = [];
			let types = [];
		}
			
		let statement = pdo->prepare(sqlStatement);
		if typeof statement == "object" {
			let statement = this->executePrepared(statement, params, types);
		}

		/**
		 * Execute the afterQuery event if an EventsManager is available
		 */
		if typeof statement == "object" {
			if typeof eventsManager == "object" {
				eventsManager->fire("db:afterQuery", this);
			}
			return new ResultPdo(this, statement, sqlStatement, bindParams, bindTypes);
		}

		return statement;
	}
```

This way we can utilize the `executePrepared` and `prepare` methods of the class.

Reply:
good!